Azure.ResourceManager.Core.Tests/ResourceTypeTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Core.Tests
+{
+    public class ResourceTypeTests
```

Comment:
Some of the tests should pass in just the resourceType instead of the entire id, we should also compare the combinations.  Right now we are only comparing full id to full id
id.CompareTo(resourceType)
resourceType.CompareTo(id)
resourceType.CompareTo(resourceType)